build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -217,8 +225,12 @@ private List<CompilationDatabaseInformation> populateObjList(IProject project, I
 						outputLocation + "", inputStrings, sourceLocation, outputLocation); //$NON-NLS-1$
 
 				IBuildMacroProvider provider = ManagedBuildManager.getBuildMacroProvider();
-				String resolvedOptionFileContents = provider.resolveValueToMakefileFormat(cmdLInfo.getCommandLine(), "", //$NON-NLS-1$
-						" ", IBuildMacroProvider.CONTEXT_FILE, //$NON-NLS-1$
+				String compilerName = CompilationDatabaseGenerator.getCompilerName(tool);
+				String commandLine = cmdLInfo.getCommandLine();
+				commandLine = commandLine.replace(compilerName, "").trim(); //$NON-NLS-1$
+				commandLine = commandLine.replaceAll("\\s+", " ").trim(); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
This seems unexpected to me - at least for corner cases. What if there is spaces in existing file names? Maybe if you can explain what you are trying to achieve with this line I will understand better.

Reply:
Code changed in latest commit, resolving.